docker/pypi/dmwm-base/Dockerfile

```diff
@@ -1,14 +1,18 @@
-FROM registry.cern.ch/cmsweb/cmsweb-base as cmsweb-base
-FROM registry.cern.ch/cmsweb/exporters as exporters
-FROM python:3.8-bullseye
+FROM registry.cern.ch/cmsweb/exporters:20250519-stable as exporters
+FROM python:3.12-slim-bookworm
 MAINTAINER Valentin Kuznetsov vkuznet@gmail.com
-RUN apt-get update && \
-    apt-get install -y curl vim libcurl4 libcurl4-openssl-dev python3-pycurl pip apache2-utils sudo less && \
-    apt-get clean
+
+# see https://docs.docker.com/build/building/best-practices/#apt-get
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl wget libcurl4 libcurl4-openssl-dev python3-pycurl \ 
```

Comment:
- Why do we need wget if we have curl, either tool is sufficient to make HTTP requests.
- Why do we need git, this implicitly invites people to log into docker image and to development which should not be promoted. I rather prefer to put effort to perform development on local laptop using plain python. That said, if we don't have it you may have by simply login to image and perform `apt-get update && apt-get install git`. Let developers to install tools within run-time environment which can be used to test/debug stuff but should not be part of image itself. Once debugging is done image can be restarted to pristine state without dev tools.
- vim is improved vi, and vi is present by default, I don't see any need for vim if we can use vi.

Reply:
I do not find vi in the default base image

```
> podman run -it --rm python:3.12-slim-bookworm bash
root@ff6a9ce1ff0e:/# vi
bash: vi: command not found
```

Reply:
then it's fine to keep vim, I didn't look explicitly nowadays what is supplied in slim image, maybe vi is no longer default editor in Debian :), or maybe slim does not need vi per se as it is very strict and small image. Either way we can install vim or do not handle it at all and leave it up to developers to install during runtime of debugging process.

Reply:
wget was used by a script to install yui and was originally provided by the python:3.8 image. i changed the script to use curl and removed wget

Reply:
regarding `git`, i understand your point, i do not care too much. i am fine either way

Reply:
since you created separate issue for clean-up images https://github.com/dmwm/WMCore/issues/12374 then I'm ok with leaving this for time being.